src/Sylius/Component/Currency/Converter/CurrencyConverter.php

```diff
@@ -11,71 +11,59 @@
 
 namespace Sylius\Component\Currency\Converter;
 
-use Sylius\Component\Currency\Model\CurrencyInterface;
-use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Currency\Model\ExchangeRateInterface;
+use Sylius\Component\Currency\Repository\ExchangeRateRepositoryInterface;
 
 /**
  * @author Paweł Jędrzejewski <pawel@sylius.org>
  */
 class CurrencyConverter implements CurrencyConverterInterface
 {
     /**
-     * @var RepositoryInterface
+     * @var ExchangeRateRepositoryInterface
      */
-    protected $currencyRepository;
+    protected $exchangeRateRepository;
 
     /**
      * @var array
      */
     private $cache;
 
     /**
-     * @param RepositoryInterface $currencyRepository
+     * @param ExchangeRateRepositoryInterface $exchangeRateRepository
      */
-    public function __construct(RepositoryInterface $currencyRepository)
+    public function __construct(ExchangeRateRepositoryInterface $exchangeRateRepository)
     {
-        $this->currencyRepository = $currencyRepository;
+        $this->exchangeRateRepository = $exchangeRateRepository;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function convertFromBase($amount, $targetCurrencyCode)
+    public function convert($amount, $sourceCurrencyCode, $targetCurrencyCode)
     {
-        $currency = $this->getCurrency($targetCurrencyCode);
+        $exchangeRate = $this->getExchangeRate($sourceCurrencyCode, $targetCurrencyCode);
 
-        if (null === $currency) {
-            throw new UnavailableCurrencyException($targetCurrencyCode);
+        if (null === $exchangeRate) {
+            return $amount;
         }
 
-        return (int) round($amount * $currency->getExchangeRate());
+        return (int) round($amount * $exchangeRate->getRatio());
```

Comment:
We have two cases here:
1. Ratio for `source` -> `target` (multiply)
2. Ratio for `target` -> `source` (divide)
```
if ($exchangeRate->getSourceCurrency()->getCode() === $sourceCurrencyCode) {
    return (int) round(amount * $exchangeRate->getRatio());
}

return (int) round(amount / $exchangeRate->getRatio());
```
This should be covered by Behat.